posthog-react-native/src/posthog-rn.ts

```diff
@@ -160,4 +168,65 @@ export class PostHog extends PostHogCore {
   initReactNativeNavigation(options: PostHogAutocaptureOptions): boolean {
     return withReactNativeNavigation(this, options)
   }
+
+  async captureNativeAppLifecycleEvents(): Promise<void> {
+    // See the other implementations for reference:
+    // ios: https://github.com/PostHog/posthog-ios/blob/3a6afc24d6bde730a19470d4e6b713f44d076ad9/PostHog/Classes/PHGPostHog.m#L140
+    // android: https://github.com/PostHog/posthog-android/blob/09940e6921bafa9e01e7d68b8c9032671a21ae73/posthog/src/main/java/com/posthog/android/PostHog.java#L278
+    // android: https://github.com/PostHog/posthog-android/blob/09940e6921bafa9e01e7d68b8c9032671a21ae73/posthog/src/main/java/com/posthog/android/PostHogActivityLifecycleCallbacks.java#L126
+
+    const prevAppBuild = this.getPersistedProperty(PostHogPersistedProperty.InstalledAppBuild)
+    const prevAppVersion = this.getPersistedProperty(PostHogPersistedProperty.InstalledAppVersion)
+    const appBuild = this._appProperties.$app_build
+    const appVersion = this._appProperties.$app_version
+
+    if (!appBuild || !appVersion) {
+      console.warn(
+        'PostHog could not track installation/update/open, as the build and version were not set. ' +
+          'This can happen if some dependencies are not installed correctly, or if you have provided' +
+          'customAppProperties but not included $app_build or $app_version.'
+      )
+    }
+    if (appBuild) {
+      if (!prevAppBuild) {
+        // new app install
+        this.capture('Application Installed', {
+          version: appVersion,
+          build: appBuild,
+        })
+      } else if (prevAppBuild !== appBuild) {
+        // app updated
+        this.capture('Application Updated', {
+          previous_version: prevAppVersion,
+          previous_build: prevAppBuild,
+          version: appVersion,
+          build: appBuild,
+        })
+      }
+    }
+
+    const initialUrl = (await Linking.getInitialURL()) ?? undefined
+
+    this.capture('Application Opened', {
+      version: appVersion,
+      build: appBuild,
+      from_background: false,
+      url: initialUrl,
+    })
+
+    AppState.addEventListener('change', (state) => {
+      if (state === 'active') {
+        this.capture('Application Opened', {
+          version: appVersion,
+          build: appBuild,
+          from_background: true,
+        })
+      } else if (state === 'background') {
+        this.capture('Application Backgrounded')
+      }
+    })
+
+    this.setPersistedProperty(PostHogPersistedProperty.InstalledAppBuild, appBuild)
```

Comment:
Given that this logic is super dependent on persisted values, maybe we need a check to make sure that we have "real" storage?

Reply:
I think we know this just based on it being a React Native app, we use `expo-file-system` or `@react-native-async-storage/async-storage` and throw if neither one is available. The user can provide their own but if they do that I don't think it's unreasonable to epxect that it's "real"

Reply:
Warning added